lib/IRGen/IRGen.cpp

```diff
@@ -385,24 +385,24 @@ static bool needsRecompile(StringRef OutputFilename, ArrayRef<uint8_t> HashData,
     StringRef SectionName;
     Section.getName(SectionName);
     if (SectionName == HashSectionName) {
-      StringRef SectionData;
-      Section.getContents(SectionData);
-      ArrayRef<uint8_t> PrevHashData(
-          reinterpret_cast<const uint8_t *>(SectionData.data()),
-          SectionData.size());
-      LLVM_DEBUG(if (PrevHashData.size() == sizeof(MD5::MD5Result)) {
-        if (DiagMutex) DiagMutex->lock();
-        SmallString<32> HashStr;
-        MD5::stringifyResult(
-            *reinterpret_cast<MD5::MD5Result *>(
+      if (llvm::Expected<llvm::StringRef> SectionData = Section.getContents()) {
```

Comment:
I think that it would be nicer if you do an early return here if the returned value is not valid (possibly reporting an error).